src/shelly_hap_light_bulb.cpp

```diff
@@ -204,6 +220,23 @@ void LightBulb::HSVtoRGBW(RGBW &rgbw) const {
   }
 }
 
+void LightBulb::ColortemperaturetoRGBW(RGBW &rgbw) const {
+  float v = cfg_->brightness / 100.0f;
+
+  // brightness and colortemperature in mired to cw, ww values
+  int temp_max = 400;
+  int temp_min = 50;
+  float temp = cfg_->colortemperature;
+  temp -= temp_min;
+  temp /= (temp_max - temp_min);
+  rgbw.ww0 = temp * v;
+  rgbw.cw0 = (1.0f - temp) * v;
+
+  // unused so far, mirror first values
+  rgbw.ww1 = rgbw.ww0;
+  rgbw.cw1 = rgbw.cw0;
```

Comment:
The CCT mode should not mirror the output, it should offer a second component to allow two separate CCT's, which if the user want can be grouped in HomeKit to one Accessory.

Reply:
Yep, this makes sense as soon as the major refactor for multiple bulbs per controller is implemented